src/domain/synthetics/trade/useTradeboxState.ts

```diff
@@ -705,10 +801,10 @@ function fallbackCollateralTokens(newState: StoredTradeOptions, marketsInfoData:
       collateralAddress = marketInfo.shortTokenAddress;
     }
 
-    set(newState, ["collaterals", marketAddress, longOrShort], collateralAddress);
+    return produce(newState, (draft) => {
+      set(draft, ["collaterals", marketAddress, longOrShort], collateralAddress);
+    });
```

Comment:
i dont really understand the benefit of using this whole creating proxy setter with immer to just set one field

Reply:
imagine doing dis without immer

Reply:
yep... it would be like:
```
return {
    ... nextState,
    collaterals: {
        ... nextState.collaterals,
        [marketAddress]: {
            ... nextState.collaterals[marketAddress],
            [longOrShort]: collateralAddress,
        },
    },
};
```

And we also need to take care of the fact that these structures might not exist (which is why we use `set`). As for me, immer greatly enhances readability.

Reply:
i undestandd the benefit of "set". I am the one who wrote it. How does immer specifically increase readability

Reply:
discussed in im